pkg/kv/kvserver/consistency_queue.go

```diff
@@ -65,22 +65,26 @@ const consistencyCheckRateMinWait = 100 * time.Millisecond
 // which can schedule checks on other nodes, e.g. a 7-node cluster with a
 // replication factor of 7 could run 7 concurrent checks on every node.
 //
-// Note that checksum calculations below Raft are not tied to the caller's
-// context, and may continue to run even after the caller has given up on them,
-// which may cause them to build up. Although we do best effort to cancel the
-// running task on the receiving end when the incoming request is aborted.
+// Note that checksum calculations below Raft are loosely tied to the caller's
+// context, and will be canceled if the caller has not joined or has given up on
+// them. This way we do best effort to avoid a build-up of abandoned tasks.
 //
 // CHECK_STATS checks do not count towards this limit, as they are cheap and the
 // DistSender will parallelize them across all ranges (notably when calling
 // crdb_internal.check_consistency()).
 const consistencyCheckAsyncConcurrency = 7
 
 // consistencyCheckAsyncTimeout is a below-Raft timeout for asynchronous
-// consistency check calculations. These are not tied to the caller's context,
-// and thus may continue to run even if the caller has given up on them, so we
-// give them an upper timeout to prevent them from running forever.
+// consistency check calculations. These are loosely tied to the caller's
+// context, and will be canceled when the caller has given up on them, but we
+// give them an upper timeout to prevent them from running forever anyway.
 const consistencyCheckAsyncTimeout = time.Hour
```

Comment:
I wonder if we should go ahead and remove this and `consistencyCheckAsyncConcurrency`? They were hacks introduced only because we didn't have proper cancellation, but now that we do (and assuming that it's reliable) I don't think we should need these crutches. `AsyncConcurrency` can also misfire, e.g. if someone is running with RF >= 9.

Reply:
I was thinking of the same, because now things don't leak. The only case when build-up is possible is when replicas restart, but this is limited to 5s. So we could either remove the concurrency limit, or make it much higher (just in case many [correlated] failures occur).

Reply:
Let's just remove it outright. A build-up is fine if it's limited to 5 seconds.

Reply:
Done.

Reply:
Ah wait. I only removed the concurrency limit (and the semaphore) so far (see the second commit). Why is it ok to remove `consistencyCheckAsyncTimeout`? The queue automatically sets a timeout?

Reply:
Done both now, see commits 2 and 3.